bundles/org.openhab.binding.guntamatic/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,596 @@
+			<property name="program">PR001</property>
+			<property name="heatCircProgram">HKx01</property>
+			<property name="wwHeat">BKx06</property>
+			<property name="extraWwHeat">ZKx06</property>
+		</properties>
+		<config-description>
+			<parameter name="hostname" type="text" required="true">
+				<context>network-address</context>
+				<label>Hostname</label>
+				<description>Hostname or IP address of the Guntamatic Heating System</description>
+			</parameter>
+			<parameter name="key" type="text" required="false">
+				<context>key</context>
+				<label>Key</label>
+				<description>Optional, but required to read protected parameters and to control the Guntamatic Heating System.
+					The
+					key needs to be reqested from Guntamatic support.</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" unit="s" min="1" required="false">
+				<label>Refresh Interval</label>
+				<description>Interval the Guntamatic Heating System is polled in seconds. Default: 60s</description>
+			</parameter>
+			<parameter name="encoding" type="text" required="false">
+				<label>Encoding</label>
+				<description>Code page used by the Guntamatic Heating System. Default: 'windows-1252'</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+
+	<channel-type id="controlProgram" advanced="false">
+		<item-type>String</item-type>
+		<label>Set Program</label>
+		<description>Set Program of the Guntamatic Heating System (OFF, NORMAL, WARMWATER, MANUAL)</description>
+		<state>
+			<options>
+				<option value="0">OFF</option>
+				<option value="1">NORMAL</option>
+				<option value="2">WARMWATER</option>
+				<option value="8">MANUAL</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="controlBoilerApproval" advanced="true">
+		<item-type>String</item-type>
+		<label>Set Boiler Approval</label>
+		<description>Set Boiler Approval of the Guntamatic Heating System (AUTO, OFF, ON)</description>
+		<state>
+			<options>
+				<option value="0">AUTO</option>
+				<option value="1">OFF</option>
+				<option value="2">ON</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="controlHeatCircProgram" advanced="true">
+		<item-type>String</item-type>
+		<label>Set Program of Heat Circle 0...8</label>
+		<description>Set Program of Heat Circle 0...8 of the Guntamatic Heating System (OFF, NORMAL, HEAT, LOWER)</description>
+		<state>
+			<options>
+				<option value="0">OFF</option>
+				<option value="1">NORMAL</option>
+				<option value="2">HEAT</option>
+				<option value="3">LOWER</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="controlWwHeat" advanced="true">
+		<item-type>String</item-type>
+		<label>Trigger Warm Wather Circle 0...2</label>
+		<description>Trigger Warm Wather Circle 0...2 of the Guntamatic Heating System (RECHARGE)</description>
+		<state>
+			<options>
+				<option value="0">RECHARGE</option>
```

Comment:
I am not sure RECHARGE is the correct English word.

Note that for all your option labels, you are not forced to use uppercase.

Reply:
> I am not sure RECHARGE is the correct English word.

I was also not sure, but this is the same term that the heating system is using.

> Note that for all your option labels, you are not forced to use uppercase.
All the other bindings that I am using, seem to have their commands in uppercase as well. Therefore I would like to keep it.

Reply:
Yes, it makes sense to have the command value in uppercase, but not necessarily its label.
In your case, the command is "0".

Reply:
Yes, the Java command value is "0", but the user should be able to call `Biostar_ControlExtraWwHeat0.sendCommand("RECHARGE")`

Reply:
I see in your code that you take the command value to call your request and then set the channel state with the result of the request. 
Just to understand, what will be the request result if your command value is "0" and if your command value is "RECHARGE" ?

Reply:
`Biostar_ControlExtraWwHeat0.sendCommand("RECHARGE")` returns `{"ack":"confirmation message"}`.
I will check the result of `Biostar_ControlExtraWwHeat0.sendCommand(0)` as soon as I have the ability to test.

Reply:
Test with "0" as parameter, not 0

Reply:
`Biostar_ControlExtraWwHeat0.sendCommand(0)`, `"0"` a well as `"RECHARGE"` lead to `"ack"`

Reply:
That is weird to use as channel state a JSON value. Generally, a binding creates a more clever state value. And a string state must be short. If too long, it will be truncated by certain UIs.